hooks/firebird/use-event-stream.tsx

```diff
@@ -105,14 +110,47 @@ const useDeploymentEventStream = (deploymentId: string) => {
       if (event instanceof MessageEvent) {
         const data: DeploymentStreamMessage = JSON.parse(event.data)
         console.log("SSE: Client received a message", data)
-
         updateDeploymentProgress(data)
         updateDeploymentStatus(data)
       }
     },
     [updateDeploymentProgress, updateDeploymentStatus],
   )
 
+  // To keep method consistent with the event stream and avoid unnecessary call errors
+  useEffect(() => {
+    if (deploymentStatus === DeploymentStatus.STARTING) {
+      setLastDeploymentStatus(DeploymentStatus.STARTING)
+    } else if (
+      deploymentStatus === DeploymentStatus.STOPPING ||
+      deploymentStatus === DeploymentStatus.STOPPED
+    ) {
+      setLastDeploymentStatus(DeploymentStatus.STOPPING)
+    } else if (deploymentStatus === DeploymentStatus.DEPLOYING) {
+      setLastDeploymentStatus(DeploymentStatus.DEPLOYING)
+    }
```

Comment:
Still need to discuss with you the role of `lastDeploymentStatus`... Reading this code, it is kind of confusing. Why do you need it?

Assuming you do need `lastDeploymentStatus`, why are you checking if `deploymentStatus` is equal to some variable and then setting it to that variable if true?

Why don't you just setLastDeploymentStatus to whatever deployment status is??

`setLastDeploymentStatus(deploymentStatus)`